core/near-network/src/lib.rs

```diff
@@ -0,0 +1,439 @@
+use ::primitives::types::AccountId;
+use ::primitives::hash::CryptoHash;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::io::{Error, ErrorKind};
+use ::tokio::net::{TcpStream, TcpListener};
+use ::tokio_codec::{Framed};
+use ::tokio::prelude::stream::SplitStream;
+use ::futures::sync::mpsc::{channel, Sender};
+use ::futures::{Stream, Future, Sink};
+use ::log::error;
+use ::serde_derive::{Serialize, Deserialize};
+use ::rand::{thread_rng, seq::IteratorRandom};
+use ::parking_lot::RwLock;
+use std::sync::Arc;
+use ::tokio_serde_cbor::Codec;
+
+#[derive(PartialEq, Eq, Hash, Clone, Debug, Serialize, Deserialize)]
+/// Information about a peer
+pub struct Peer {
```

Comment:
If nothing outside network is going to be using this struct then it does not have to be `pub`, and there is no need for constructor.

Reply:
Fixed in d9e5a3f